src/lTerm_draw.ml

```diff
@@ -586,3 +586,10 @@ let draw_frame ctx rect ?style connection =
   draw_piece ctx (rect.row1 + 0) (rect.col2 - 1) ?style { top = Blank; bottom = connection; left = connection; right = Blank };
   draw_piece ctx (rect.row2 - 1) (rect.col2 - 1) ?style { top = connection; bottom = Blank; left = connection; right = Blank };
   draw_piece ctx (rect.row2 - 1) (rect.col1 + 0) ?style { top = connection; bottom = Blank; left = Blank; right = connection }
+
+let draw_frame_labelled ctx rect ?style ?(alignment=H_align_left) label connection = 
+  draw_frame ctx rect ?style connection;
+  try 
+    let ctx = sub ctx { row1=0; row2=1; col1=1; col2=rect.col2-1 } in
+    draw_string_aligned ctx rect.row1 alignment label
```

Comment:
I think the rectangle given to `sub` should be: `{row1=rect.row1;row2=rect.row1+1;col1=rect.col1+1;col2=rect.col2-1}`

the `draw_string_aligned` should be given row 0 as it is relative to the new context